packages/secure_storage/amplify_secure_storage_dart/lib/src/amplify_secure_storage_dart.dart

```diff
@@ -38,6 +43,58 @@ class AmplifySecureStorageDart extends AmplifySecureStorageInterface
     with AmplifySecureStorageDartMixin {
   /// {@macro amplify_secure_storage_dart.amplify_secure_storage_dart}
   AmplifySecureStorageDart({
-    required AmplifySecureStorageConfig config,
-  }) : super(config: config);
+    required super.config,
+  });
+}
+
+/// {@template amplify_secure_storage_dart.amplify_secure_storage_worker}
+/// A `package:worker_bee` variant of [AmplifySecureStorageDart] which runs
+/// storage operations in a background thread.
+/// {@endtemplate}
+class AmplifySecureStorageWorker extends AmplifySecureStorageInterface {
+  /// {@macro amplify_secure_storage_dart.amplify_secure_storage_worker}
+  AmplifySecureStorageWorker({
+    required super.config,
+  });
+
+  late final SecureStorageWorker _worker;
+  final _workerMemo = AsyncMemoizer<void>();
+
+  @override
+  Future<void> init() => _workerMemo.runOnce(() async {
```

Comment:
I am seeing the following exception when running the Auth example app and the Secure Storage example app on Mac.

```
LateError (LateInitializationError: Field '_worker@619435580' has not been initialized.)
```

For the Secure Storage example app that might be resolved by calling `storage.init()`. It looks like that was already added to the auth example though, I am still seeing that. I am not sure if it is not waiting for it.

It might make sense to confirm init() has completed before calling read/write/delete. `_workerMemo` is already an `AsyncMemoizer`, so I think you could just add `await init()` before each method. I am not super familiar with `AsyncMemoizer` so maybe it isn't quire that simple.

Reply:
Ah okay, yeah that's probably easiest. Will do!